board/safety/safety_toyota.h

```diff
@@ -46,19 +48,32 @@ static void toyota_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
     // increase torque_meas by 1 to be conservative on rounding
     toyota_torque_meas.min--;
     toyota_torque_meas.max++;
-
   }
 
   // enter controls on rising edge of ACC, exit controls on ACC off
   if ((to_push->RIR>>21) == 0x1D2) {
     // 5th bit is CRUISE_ACTIVE
     int cruise_engaged = to_push->RDLR & 0x20;
-    if (cruise_engaged && !toyota_cruise_engaged_last) {
-      controls_allowed = 1;
-    } else if (!cruise_engaged) {
+    // 4th bit is GAS_RELEASED
+    int gas = !(to_push->RDLR & 0x10);
+    if (!cruise_engaged || (gas && !toyota_gas_prev && !gas_interceptor_detected)) {
       controls_allowed = 0;
+    } else if (cruise_engaged && !toyota_cruise_engaged_last) {
+      controls_allowed = 1;
     }
     toyota_cruise_engaged_last = cruise_engaged;
+    toyota_gas_prev = gas;
+  }
+
+  // exit controls on rising edge of gas press if interceptor (0x201)
+  if ((to_push->RIR>>21) == 0x201) {
+    gas_interceptor_detected = 1;
+    int gas_interceptor = ((to_push->RDLR & 0xFF) << 8) | ((to_push->RDLR & 0xFF00) >> 8);
+    if ((gas_interceptor > TOYOTA_GAS_INTERCEPTOR_THRESHOLD) &&
+        (toyota_gas_interceptor_prev <= TOYOTA_GAS_INTERCEPTOR_THRESHOLD)) {
+      controls_allowed = 0;
+    }
+    toyota_gas_interceptor_prev = gas_interceptor;
```

Comment:
gas_interceptor is global, but honda_gas_interceptor_prev and toyota_gas_interceptor_prev are specific to the file. Why?

Reply:
gas_interceptor is sent in the health packet so should be global. I wanted to limit the amount of global vars, so kept gas_interceptor_prev file specific. I can make them global too, it will simplify both safety code and tests.

Reply:
If they have the exact same meaning I think they can be shared in a "safety.h global"

Reply:
well, they are global anyway, just a matter of where they appear. The interceptor itself is abstracted from the specific car, so I'll just move gas_interceptor_prev to safety.h